crates/swc_atoms/src/lib.rs

```diff
@@ -190,9 +219,9 @@ impl Ord for Atom {
 impl PartialEq for Atom {
     fn eq(&self, other: &Self) -> bool {
         // Fast path
```

Comment:
@kdy1 After supporting `Atom::static`, I'm not sure the fast path will work as before. To keep the logic the same as before, we need to call `match` twice. I'm going to remove the code, in case I miss something, what are your opinions?

Reply:
Also, should `Atom` promise `assert(Atom::Arc("foo") == Atom::Static("foo"))`?

Reply:
I think it's fine, but `assert!(Atom::Arc("foo") == Atom::Static("foo"))` should work